retrieve/retrieve.go

```diff
@@ -0,0 +1,103 @@
+package retrieve
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/joshuatcasey/collections"
+	"github.com/joshuatcasey/libdependency"
+	"github.com/joshuatcasey/libdependency/versionology"
+	"github.com/joshuatcasey/libdependency/workflows"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"golang.org/x/exp/slices"
+)
+
+type GenerateMetadataFunc func(version versionology.HasVersion) (cargo.ConfigMetadataDependency, error)
```

Comment:
You may recall that there will not be a `targets.json` file. It will be part of the metadata.json file as a field. 

We didn't end up adding the `target` field to the cargo dependency structure, because we don't want it in the `buildpack.toml`, but we do want it in the `metadata.json`.

[The dependency workflow](https://github.com/paketo-buildpacks/github-config/pull/537/files#diff-bfc823f104430e9254a88f78b3733124280dd323c73abc2e5c18673a6fc801aaR128) will use it directly from the `metadata.json`.

Reply:
to be fixed with next push